Source/WebInspectorUI/UserInterface/Views/ConsoleDrawer.js

```diff
@@ -154,7 +154,7 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
     _updateDrawerHeight(height)
     {
         const minimumHeight = 64;
-        const maximumHeight = this.element.parentNode.offsetHeight - WI.TabBrowser.MinimumHeight - WI.QuickConsole.MinimumHeight;
+        const maximumHeight = this.element.parentNode.offsetHeight - WI.TabBrowser.MinimumHeight - WI.quickConsole.element.offsetHeight;
```

Comment:
can you explain more the purpose of this specific part?

Style: i think this is technically a layering violation, so we should figure out a different way to access this
Style: this should be `let` as the (computed) value changes each time it's created (i.e. it's not truly constant)

Reply:
> can you explain more the purpose of this specific part?

When resizing the Console drawer, we need to account for the min-height of the Tab content area (`WI.TabBrowser.MinimumHeight`) and for the height of the Console prompt. 

When there is multi-line input in the Console prompt, the _intent_ is to allow it to grow up to `33vh`. 
The drawer, prompt and Tab content area are all flex items so the fixed height of the drawer reduces from the available space to be distributed to the other items.

Without this change, the Console drawer can steal from the prompt's height. 
Here's a demo of the behavior (left: this patch, right: current Safari):

https://github.com/user-attachments/assets/a3874df1-a986-4836-b476-1200c797dd21


> Style: i think this is technically a layering violation, so we should figure out a different way to access this

Yeah, it's not ideal but they are semantically related. Do you have a suggestion how to separate?

Reply:
ok yeah i can see why that's useful/required

i was specifically referring to accessing the `WI.quickConsole` instance

it looks like we already calculate the `quickConsoleHeight` inside `WI.ConsoleDrawer.prototype._consoleResizerMouseDown`.  rather than fetch the `offsetHeight` on _every_ user action, could we save that value to a variable and use that instead?

Reply:
> it looks like we already calculate the quickConsoleHeight inside WI.ConsoleDrawer.prototype._consoleResizerMouseDown. rather than fetch the offsetHeight on every user action, could we save that value to a variable and use that instead?

I tried this and I have some concerns about it.

```diff
diff --git a/Source/WebInspectorUI/UserInterface/Views/ConsoleDrawer.js b/Source/WebInspectorUI/UserInterface/Views/ConsoleDrawer.js
index 3940ba737761..67caa4002819 100644
--- a/Source/WebInspectorUI/UserInterface/Views/ConsoleDrawer.js
+++ b/Source/WebInspectorUI/UserInterface/Views/ConsoleDrawer.js
@@ -89,6 +89,13 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
         return this.element.offsetHeight;
     }
 
+    get quickConsoleHeight()
+    {
+        this._quickConsoleHeight ??= window.innerHeight - (this.element.totalOffsetTop + this.height);
+
+        return this._quickConsoleHeight;
+    }
+
     // Protected
 
     attached()
@@ -105,6 +112,8 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
         if (this._collapsed)
             return;
 
+        this._quickConsoleHeight = null;
+
         let height = this.height;
         this._restoreDrawerHeight();
 
@@ -124,8 +133,7 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
             return;
 
         let resizerElement = event.target;
-        let quickConsoleHeight = window.innerHeight - (this.element.totalOffsetTop + this.height);
-        let mouseOffset = quickConsoleHeight - (event.pageY - resizerElement.totalOffsetTop);
+        let mouseOffset = this.quickConsoleHeight - (event.pageY - resizerElement.totalOffsetTop);
 
         function dockedResizerDrag(event)
         {
@@ -140,6 +148,8 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
             if (event.button !== 0)
                 return;
 
+            this._quickConsoleHeight = null;
+
             WI.elementDragEnd(event);
         }
 
@@ -154,7 +164,7 @@ WI.ConsoleDrawer = class ConsoleDrawer extends WI.ContentBrowser
     _updateDrawerHeight(height)
     {
         const minimumHeight = 64;
-        let maximumHeight = this.element.parentNode.offsetHeight - WI.TabBrowser.MinimumHeight - WI.quickConsole.element.offsetHeight;
+        let maximumHeight = this.element.parentNode.offsetHeight - WI.TabBrowser.MinimumHeight - this.quickConsoleHeight;
 
         let heightCSSValue = Number.constrain(height, minimumHeight, maximumHeight) + "px";
         if (this.element.style.height === heightCSSValue)
```

We need to compute the value of `this._quickConsoleHeight` independent of `WI.ConsoleDrawer._consoleResizerMouseDown()` because it is needed when restoring the drawer height when opening Web Inspector in `WI.ConsoleDrawer._restoreDrawerHeight()`.

We get an incorrect size if we lazy compute and cache the value on the first call:
`this._quickConsoleHeight ??= window.innerHeight - (this.element.totalOffsetTop + this.height);`

We _assume_ this is the height of the quick console, but during the initial layout stages, `this.element.totalOffsetTop + this.height` will be `0` so we end up with a wrong value, a quick console the full height of viewport. We have to clear this value both in `WI.ConsoleDrawer.sizeDidChange()`, as the view structure is laid out, and after stopping dragging the console drawer resizer.

I think this makes it unnecessarily complex to reason about is prone to bugs in the future.

I agree that it's technically a layering violation, but I'd rather keep the direct access to `WI.quickConsole.element.offsetHeight` for clarity and for correctness.

Reply:
@dcrousso do you agree that we can keep `WI.quickConsole.element.offsetHeight` for clarity and for correctness?

Reply:
my concern with this is that there's already some performance issues with resizing the Console (and Web Inspector in general) and i'd hate to make the situation worse by accessing values that force a relayout (if dirty)

is there truly no way of grabbing this value just once right when dragging starts (or maybe even shortly thereafter) so that we dont have to worry about perf?

what if instead of caching it in a `_quickConsoleHeight` property, we newly compute it each time in `_consoleResizerMouseDown` and `_restoreDrawerHeight` to then pass it as an argument to `_updateDrawerHeight`?

Reply:
> what if instead of caching it in a _quickConsoleHeight property, we newly compute it each time in _consoleResizerMouseDown and _restoreDrawerHeight to then pass it as an argument to _updateDrawerHeight?

That's a reasonable approach 👍  
Added in the latest patch.

> my concern with this is that there's already some performance issues with resizing the Console (and Web Inspector in general) and i'd hate to make the situation worse by accessing values that force a relayout (if dirty)

Agreed. 

The issue that affects performance most here seems to be the resizing the Web Inspector window which will trigger `WI.ConsoleDrawer.sizeDidChange()` frequently and that always recalculates the available space for the drawer in `_restoreDrawerHeight()`. 

But that's a separate issue, ideally best resolved in CSS.